### PR TITLE
Comment a couple of high-level data structs

### PR DESCRIPTION
**Pull request by PhilipDaniels**
I am attempting to understand how Lapce works in more detail, and I thought it might be a good idea to start commenting some of the 'Data' structures.

If this is OK, it is my intention to go further down the tree.

### CODE REVIEW

File: lapce-data/src/data.rs [hunk 1]
```diff
@@ -156,20 +167,44 @@ impl LapceData {
     }
 }
 
+/// `LapceWindowData` is the application model for a top-level window,
+/// that is a window which can be independently moved around and
+/// resized using your window manager. Normally Lapce has only one
+/// top-level window, but new ones can be created using the "New Window"
+/// command.
+/// 
+/// Each window has its own collection of tabs (again, there is normally
```
Comment (bugadani) [inline lapce-data/src/data.rs:176]: You might want to emphasize here as well that these tabs are not the commonly used editor tabs.

File: lapce-data/src/data.rs [hunk 2]
```diff
@@ -156,20 +167,44 @@ impl LapceData {
...
+/// only one tab), size, position etc. and `Arc` references to state that is
+/// common to this instance of Lapce, such as configuration and the
+/// keymap setup.
 #[derive(Clone)]
 pub struct LapceWindowData {
+    /// The unique identifier for the Window. Assigned by Druid.
     pub window_id: WindowId,
+    /// The set of tabs within the window. These tabs are high-level
+    /// groupings, in particular they are not **editor tabs**, which are
+    /// lower down the hierarchy.
```
Comment (bugadani) [inline lapce-data/src/data.rs:186]: For reference, would you think it made sense to link to the editor tab data struct?

**Review by bugadani — COMMENTED** [review]
This is a great start, thank you!

File: lapce-data/src/data.rs (hunk 1 above)
Comment (PhilipDaniels) [reply lapce-data/src/data.rs]: Done.

File: lapce-data/src/data.rs (hunk 2 above)
Comment (PhilipDaniels) [reply lapce-data/src/data.rs]: Done.

**Comment by PhilipDaniels** [comment]
Couple of comments for further consideration

1. I am thinking of renaming `LapceTabData` to `LapceWindowTabData` so that it contrasts better with `LapceEditorTabData`.
2. It would be really great if the docs could include some pictures of the things they are referring to - this would clarify things a lot in my opinion, half of this text could probably be deleted. (It would also be really great for theme settings.)

Looking around, rustdoc can't natively support images yet ([issue](https://github.com/rust-lang/rust/issues/32104)). It can support URLs to images hosted somewhere else on the internet, but I don't want to do that as such links inevitably suffer bitrot. I did find an interesting crate https://docs.rs/embed-doc-image-showcase/latest/embed_doc_image_showcase/ which encodes images as base64 strings and embeds them directly into the generated HTML. Basically there would be a directory in the crate root(s) with a bunch of (small) pngs which could be referred to from the doc comments. That way everything is git-versioned and self-contained.

@dzhou121 We had a short conversation on Discord about documenting things about a week ago...my latest thoughts here. For now, I strongly favour documentation that is close to the code, rather than on a wiki or a separate website.

**Review by bugadani — APPROVED** [review]
Thanks! For whatever reason I'm not sure "window tab" is the best term, but it's clear enough.

**Comment by bugadani** [comment]
> images hosted somewhere else

Images in an asset folder next to the source works decently well. We should open an issue about this, I think.

**Comment by PhilipDaniels** [comment]
> > images hosted somewhere else
> 
> Images in an asset folder next to the source works decently well. We should open an issue about this, I think.

I'll make a test PR tomorrow, see what it looks like.

**Pull request merged**